httpcore/_async/http_proxy.py

```diff
@@ -134,7 +137,13 @@ def __init__(
             ] + self._proxy_headers
 
     def create_connection(self, origin: Origin) -> AsyncConnectionInterface:
-        if origin.scheme == b"http":
+        is_tls = origin.scheme == b"https"
+
+        forwarable = not (self._mode & ProxyMode.HTTP_TUNNEL)
+        if is_tls:
+            forwarable = bool(self._mode & ProxyMode.HTTPS_FORWARD)
```

Comment:
Let's just use `is` operator in such cases

For example
```python
forwardable = not (self._mode is ProxyMode.HTTP_TUNNEL)
```

Reply:
Consider when `mode=ProxyMode.HTTP_TUNNEL | ProxyMode.HTTPS_FORWARD`

Reply:
Is there any differences between `mode=ProxyMode.HTTP_TUNNEL | ProxyMode.HTTPS_FORWARD`
and `mode=ProxyMode.DEFAULT`?

Reply:
Yes obviously, `ProxyMode.HTTP_TUNNEL | ProxyMode.HTTPS_FORWARD` would reverse the defaults.
Here is all options:


`ProxyMode.DEFAULT`:
HTTP dest: forwarding
HTTPS dest: tunneling

`ProxyMode.HTTP_TUNNEL | ProxyMode.HTTPS_FORWARD`:
HTTP dest: tunneling
HTTPS dest: forwarding

`ProxyMode.HTTP_TUNNEL`:
HTTP dest: tunneling
HTTPS dest: tunneling

`ProxyMode.HTTPS_FORWARD`:
HTTP dest: forwarding
HTTPS dest: forwarding